src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -44,6 +45,11 @@ public TelemetryClient(TelemetryConfiguration configuration)
             }
 
             this.configuration = configuration;
+
+            if (this.configuration.TelemetryChannel == null)
```

Comment:
why do we need this check at all? It wouldn't protect from failure in `Track` if customer set it to `null` later.